### DISCUSSION TITLE
Frequent ID swapping (How to improve proofreading efficiency in this case?)

### DISCUSSION THREAD

**Discussion by sdfclaj123** [talmolab/sleap#1857] [Help!]
We have already trained a good animal model and it worked well when we ran inferences on two test videos (one 2-minute video and one 10-minute video; 30fps; 12 flies in each video): accurately located each fly and very minimal ID swapping. However, when we ran inferences on the experiment videos (exact same setup as the test videos), we started to have a lot of ID swapping when one fly is chasing another. These flies are not really interacting (no body parts overlapped) and ID swapping does not only occurs when they are on the side but also when they are in perfectly flat poses. What would be the reason for this? What can we do to improve? As we have hundreds of videos to process, this ID swapping issue is very problematic. 
<img width="742" alt="SLEAP_example" src="https://github.com/talmolab/sleap/assets/144021845/1fa48534-d684-4d80-bc50-6c5d3a6b2090">

**Comment by talmo** [comment]
Hi @sdfclaj123,

Apologies for the delay!!

Since the identity tracking isn't learnable, it shouldn't be impacted by a domain gap/failure to generalize on new videos, so if the poses are still looking good on the new data, the identity tracking should work just as well.

Do you mind sharing the tracking parameters or CLI command that you're using?

Could you also share a clip of when an ID switch happens on the new data? (You can hold shift while dragging across the seekbar to select a clip, then render a video with the poses from the **View** menu.)

Cheers,

Talmo

**Reply by sdfclaj123** [reply]
Sorry for the late response! 

This would be the setting for running inference. We never get it working with "flow" so we use "simple". 
![Run_Inference_Para](https://github.com/user-attachments/assets/d22b0b65-588c-4abd-8664-69331273cfd3)

Here is one example clip of the ID swapping (top left arena, the never-ending swapping of the two instances green and purple)

![ID_swapping_example_top_left](https://github.com/user-attachments/assets/8d9f16c1-6e23-4faa-95cf-3e753ef7cc39)

Is there something we can try?

**Reply by talmo** [reply]
Hi @sdfclaj123,

Thanks for sharing the clip!

Indeed that looks problematic and unexpected.

A few questions:

1. What happens when you turn the tracking window down to 3?
2. Have you tried using the max instead of the robust quantile?
3. Disabling connect single track break?
4. Using instance similarity instead of IOU?
5. Would you mind [sharing the SLP file and video](https://forms.gle/d3eftG7mu1ooZbo18) with us so we can do some troubleshooting on our end?

Cheers,

Talmo
